linux/keyman-config/keyman_config/keyboard_details.py

```diff
@@ -36,11 +36,12 @@ class KeyboardDetailsView(Gtk.Dialog):
     # TODO clean up file once have what we want
     def __init__(self, parent, kmp):
         self.kmp = kmp
+        name = secure_lookup(kmp, 'name')
         # kmp has name, version, packageID, area
-        if "keyboard" in self.kmp["name"].lower():
-            wintitle = self.kmp["name"]
+        if name and "keyboard" in name.lower():
+            wintitle = name
         else:
-            wintitle = _("{name} keyboard").format(name=self.kmp["name"])
+            wintitle = _("{name} keyboard").format(name=name)
```

Comment:
if name is null here then shouldn't we use the basename of the package instead?

Reply:
Done. We do set `packageID` based on the basename of the package, so we can assume `self.kmp['packageID']` to exist.

Reply:
Cool. Which technically means we could make line 41 `if "keyboard" in name.lower():` because we know it will always be a string.